tests/integration/validate/test_validate_command.py

```diff
@@ -0,0 +1,68 @@
+"""
+Integration tests for sam validate
+"""
+
+import os
+import re
+from enum import Enum, auto
+from pathlib import Path
+from typing import List, Optional
+from unittest import TestCase
+
+from parameterized import parameterized
+from tests.testing_utils import run_command
+
+
+class TemplateFileTypes(Enum):
+    JSON = auto()
+    YAML = auto()
+
+
+class TestValidate(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.patterns = {
+            TemplateFileTypes.JSON: re.compile(r"^/.+/template[.]json is a valid SAM Template$"),
+            TemplateFileTypes.YAML: re.compile(r"^/.+/template[.]yaml is a valid SAM Template$"),
+        }
+
+    @staticmethod
+    def base_command() -> str:
+        return "samdev" if os.getenv("SAM_CLI_DEV") else "sam"
+
+    def command_list(
+        self,
+        template_file: Optional[Path] = None,
+        profile: Optional[str] = None,
+        region: Optional[str] = None,
+        config_file: Optional[Path] = None,
+    ) -> List[str]:
+        command_list = [self.base_command(), "validate"]
+        if template_file:
+            command_list += ["--template-file", str(template_file)]
+        if profile:
+            command_list += ["--profile", profile]
+        if region:
+            command_list += ["--region", region]
+        if config_file:
+            command_list = ["--config_file", str(config_file)]
+        return command_list
+
+    @parameterized.expand(
+        [
+            ("default_yaml", TemplateFileTypes.YAML),  # project with template.yaml
+            ("default_json", TemplateFileTypes.JSON),  # project with template.json
+            ("multiple_files", TemplateFileTypes.YAML),  # project with both template.yaml and template.json
+            (
+                "with_build",
+                TemplateFileTypes.JSON,
+            ),  # project with template.json and standard build directory .aws-sam/build/template.yaml
+        ]
+    )
+    def test_default_template(self, relative_folder: str, expected_file: TemplateFileTypes):
+        cwd = f"tests/integration/testdata/validate/{relative_folder}"
```

Comment:
Can we use this way to define the testdata path? So that integration tests can be started from any directory (also using `/` might not work on Windows)

https://github.com/aws/aws-sam-cli/blob/283e546097f7520da740560de8a963ea387098dd/tests/integration/buildcmd/test_build_cmd.py#L1467

Reply:
Agree, that's more bullet proof. Thanks! Though direct slash `/` works for windows path as well

Reply:
oh really? that's cool, does it work in normal CMD as well, apart from WSL?

Reply:
it works in PowerShell not sure about an archaic cmd though 😄 I haven't been using cmd for a long time. But in this case it will work because python supports both `\\` and `/` for windows paths